docs/pages/admin-guides/access-controls/guides/joining-sessions.mdx

```diff
@@ -1,52 +1,151 @@
-description: Describes the purpose of moderated sessions and how to configure roles to support moderated sessions in a Teleport cluster.
+title: Joining Sessions
+description: Describes shared sessions and how to configure roles to support joining sessions in a Teleport cluster.
 keywords:
   - SSH
   - Kubernetes
   - session moderation
   - audit
 ---
 
-Moderated sessions allow you to define requirements for other users to be present
-in an active server or Kubernetes session started by another user. Depending on the
-requirements you specify, users who are allowed to join other users' sessions can be
-granted permission to do the following:
+Teleport allows multiple users to join the same SSH or `kubectl exec` session.
+Session joining can be performed via the web UI's Active Sessions page, or by
+using the `tsh join` command.
 
-- Observe another user's session in real time.
-- Participate interactively in another user's session.
-- Terminate another user's session at will.
+## Participant modes
+
+When joining a session, users can be in one of three participant modes:
+
+- **Observer**: Can only view the session.
+- **Peer**: Can view the session and interact with it as if they were the session owner.
+- **Moderator**: (Enterprise only) Can view the session and terminate it.
+
+Users can join SSH sessions from the command line or from the Teleport web UI,
+but Kubernetes sessions can only be joined from the command line.
+
+The web UI forces users to select a join mode prior to joining. If you join a
+session with `tsh join` or `tsh kube join`, you can specify a participant mode
+with the `--mode <mode>` command-line option, where `<mode>` is `peer`,
+`moderator`, or `observer`. The default participant mode is `observer`.
+
+You can leave a session with the shortcut `^c` (Control + c) while in observer or
+moderator mode. In moderator mode, you can also forcefully terminate the session
+at any point in time by pressing `t`.
+
+## Access controls
+
+You can use the `join_sessions` field of a role to specify the sessions users
+can join and under what conditions they can join a session. For example, the
+following role allows users to join both SSH and Kubvernetes sessions started by
+users with the `prod-access` role and to join the session as a moderator or an
+observer:
+
+```yaml
+kind: role
+metadata:
+  name: allow-session-join
+version: v7
+spec:
+  allow:
+    join_sessions:
+      - name: Join prod sessions
+        roles : ['prod-access']
+        kinds: ['k8s', 'ssh']
+        modes: ['moderator', 'observer']
+```
+
+Users who are assigned a role with a `join_sessions` allow policy are
+implicitly allowed to list the sessions that the policy gives them permission
+to join. If there's a `deny` rule that prevents listing sessions, the
+`join_sessions` policy overrides the `deny` rule for the sessions the
+policy allows the user to join. Outside of this exception for joining
+sessions, `deny` statements take precedent.
```

Comment:
What exception? Is this just to say that listing sessions is not required to join one knowing its ID, given the appropriate permissions to do so?

Reply:
This is saying that existing deny rules on `session_tracker` _do_ work, unless the user is allowed to join the session, then the deny doesn't matter.

That's the intent at least. A quick look at the code seems to indicate this is what should happen, but I'll confirm.

Reply:
Confirmed (and wrote some extra tests) in #51314.

Explicit deny on `session_tracker` works unless you have permission to join the session, then we show you the tracker.

Explicit deny on `ssh_session` works, and denies you _even if you have permissions to join_ (for legacy reasons).